media-capabilities/index.js

```diff
@@ -17,31 +17,28 @@ import { useState, useEffect } from 'react';
 
 const supported = typeof window !== 'undefined' && 'mediaCapabilities' in navigator;
 
-const useMediaCapabilities = (mediaConfig, initialMediaCapabilities = {}) => {
-  initialMediaCapabilities = {
+const useMediaCapabilitiesDecodingInfo = (mediaDecodingConfig, initialMediaCapabilitiesInfo = {}) => {
+  initialMediaCapabilitiesInfo = {
     supported,
-    ...initialMediaCapabilities
+    ...initialMediaCapabilitiesInfo
   };
 
-  const [mediaCapabilities, setMediaCapabilities] = useState(initialMediaCapabilities);
-
-  const updateMediaCapabilities = mediaCapabilities => {
-    setMediaCapabilities(mediaCapabilities);
-  };
+  const [mediaCapabilitiesInfo, setMediaCapabilitiesInfo] = useState(initialMediaCapabilitiesInfo);
 
   useEffect(() => {
-    if (supported && !!mediaConfig) {
-      navigator
-        .mediaCapabilities
-        .decodingInfo(mediaConfig)
-        .then(updateMediaCapabilities);
+    if (supported && !!mediaDecodingConfig) {
+      try {
+        navigator
+          .mediaCapabilities
+          .decodingInfo(mediaDecodingConfig)
+          .then(setMediaCapabilitiesInfo);
+      } catch (error) { 
+        console.error(error);
+      }
```

Comment:
true, the code didn't seem to block but indeed we need to be using then/catch for this. Good catch!

Reply:
Perhaps it's tricky to use async/await inside useEffect because async callback is not supported in useEffect. Am I correct?

I have no objection to going ahead with then/catch.

Reply:
We could do it here by creating a separate async function and calling that async function in the useEffect function
```
useEffect(() => {
	(async function() {
        try {
			let mediaCapabilitiesInfo = await navigator
        	.mediaCapabilities
        	.decodingInfo(mediaDecodingConfig)
  		  	
			setMediaCapabilitiesInfo(mediaCapabilitiesInfo)
  		} catch(err) {
    		console.error(err)
  		}
    })();
});
```
but it makes it slightly less readable and in this codebase I would have to add the runtime transform babel plugin to make it work with await/async.

Reply:
and yes you are correct, the useEffect doesn't work when given an async function directly

Reply:
then/catch would sound good to me.